src/containers/find-offer/offer-container/OfferContainer.tsx

```diff
@@ -0,0 +1,51 @@
+import { FC } from 'react'
+import Box from '@mui/material/Box'
+import Grid from '@mui/material/Grid'
+import { CardsViewTypes, OfferResponse, CardsViewEnums } from '~/types'
+import OfferCardSquare from '~/containers/find-offer/offer-card-square/OfferCardSquare'
+import OfferCard from '~/components/offer-card/OfferCard'
+
+interface OfferContainerProps {
+  viewMode: CardsViewTypes
+  offerCards: OfferResponse[]
+}
+
+const OfferContainer: FC<OfferContainerProps> = ({ viewMode, offerCards }) => {
+  const onBookmarkClick = (id: string) => {
+    console.log(id)
+  }
+
+  const columnNumber = viewMode === CardsViewEnums.Grid ? 12 : 1
+  const cardsArray =
+    viewMode === CardsViewEnums.Grid
+      ? offerCards.map((el) => (
```

Comment:
dublicate of code, check for viewMode inside grid and render component you need or move this check to variable and pass this variable inside Grid

Reply:
Discussed.

Reply:
i think Tolik is right, you should do:
offerCards.map((el) => (
          <Grid item key={el._id} sm={4}>
       viewMode === CardsViewEnums.Grid ?  <OfferCard
              offer={el}
              onBookmarkClick={() => onBookmarkClick(el._id)}
            /> :             <OfferCard
              offer={el}
              onBookmarkClick={() => onBookmarkClick(el._id)}
            />
          </Grid>

Reply:
agree with Mykhailo, also, you can move this check to variable and pass it inside grid
const offerCard = viewMode === CardsViewEnums.Grid ? <OfferCard
offer={el}
onBookmarkClick={() => onBookmarkClick(el._id)}
/> : <OfferCard
offer={el}
onBookmarkClick={() => onBookmarkClick(el._id)}
/>

offerCards.map((el) => (
<Grid item key={el._id} sm={4}>
{offerCard}
</Grid>\

Reply:
We have already reviewed this part of the code with Anatolii. Grid item requires filling in the "key" field. So, Grid item must be inside the array.

Reply:
yeah, but here i leave Grid in map

Reply:
Done.